lib/SIL/SILProfiler.cpp

```diff
@@ -1017,6 +1056,11 @@ struct CoverageMapping : public ASTWalker {
     if (skipExpr(E))
       return {true, E};
 
+    // Profiling for closures should be handled separately. Do not visit
```

Comment:
You might want to consider creating a new CoverageASTWalker or similar to extract some of the common logic between your ASTWalker implementations.